packages/search/src/search.tsx

```diff
@@ -84,366 +91,389 @@ type Props = {
-		hasFocus: this.props.autoFocus ?? false,
-	};
+		onSearchChange?.( keyword );
+	}, [ doSearch, keyword, onSearchChange ] );
 
-	openSearch = (
-		event:
-			| MouseEvent< HTMLButtonElement | HTMLInputElement >
-			| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
-	): void => {
+	const openSearch = ( event: KeyboardOrMouseEvent ): void => {
 		event.preventDefault();
-		this.setState( {
-			keyword: '',
-			isOpen: true,
-		} );
-		this.props.onSearchOpen?.( event );
+
+		setKeyword( '' );
+		setIsOpen( true );
+
+		focus();
+		// no need to call `onSearchOpen` as it will be called by `onFocus` once the searcbox is focused
 		// prevent outlines around the open icon after being clicked
-		this.openIcon.current?.blur();
-		this.props.recordEvent?.( 'Clicked Open Search' );
+		openIcon.current?.blur();
+		recordEvent?.( 'Clicked Open Search' );
 	};
 
-	closeSearch = (
-		event:
-			| MouseEvent< HTMLButtonElement | HTMLInputElement >
-			| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
-	): void => {
+	const closeSearch = ( event: KeyboardOrMouseEvent ): void => {
 		event.preventDefault();
 
-		if ( this.props.disabled ) {
+		if ( disabled ) {
 			return;
 		}
 
-		this.setState( {
-			keyword: '',
-			isOpen: false,
-		} );
+		setKeyword( '' );
+		setIsOpen( false );
 
-		if ( this.searchInput.current ) {
-			this.searchInput.current.value = ''; // will not trigger onChange
+		if ( searchInput.current ) {
+			searchInput.current.value = ''; // will not trigger onChange
 		}
 
-		if ( this.props.pinned ) {
-			this.searchInput.current?.blur();
-			this.openIcon.current?.focus();
+		if ( pinned ) {
+			searchInput.current?.blur();
+			openIcon.current?.focus();
 		} else {
-			this.searchInput.current?.focus();
+			searchInput.current?.focus();
 		}
 
-		this.props.onSearchClose?.( event );
+		onSearchClose?.( event );
 
-		this.props.recordEvent?.( 'Clicked Close Search' );
+		recordEvent?.( 'Clicked Close Search' );
 	};
 
-	closeListener = keyListener( this.closeSearch );
-	openListener = keyListener( this.openSearch );
+	const closeListener = keyListener( closeSearch );
+	const openListener = keyListener( openSearch );
 
-	componentDidUpdate( _: Props, prevState: State ): void {
-		this.scrollOverlay();
-		// Focus if the search box was opened
-		if ( this.state.isOpen && ! prevState.isOpen ) {
-			this.focus();
-		}
-
-		if ( this.state.keyword === prevState.keyword ) {
-			return;
-		}
-
-		// if there's a keyword change: trigger search
-		if ( this.state.keyword ) {
-			this.onSearch( this.state.keyword );
-		} else {
-			// explicitly bypass debouncing when there is no keyword is empty
-			if ( this.props.delaySearch ) {
-				// Cancel any pending debounce
-				this.onSearch.cancel?.();
-			}
-
-			this.props.onSearch( this.state.keyword );
-		}
-
-		this.props.onSearchChange?.( this.state.keyword );
-	}
-
-	scrollOverlay = (): void => {
-		this.overlay &&
+	const scrollOverlay = (): void => {
+		overlay &&
```

Comment:
This `overlay` check is redundant, and was even in the original class component. The variable always has a `{ current }` value. It's only the `overlay.current` field that can be `null`.